common/model-views.cpp

```diff
@@ -1951,13 +1911,31 @@ namespace rs2
         return results;
     }
 
-    std::vector<stream_profile> subdevice_model::get_selected_profiles()
+    bool is_resolution_max(const std::vector<std::pair<int, int>>& res_values, int width, int height, const std::pair<int, int>& calibration_res)
+    {
+        auto size = res_values.size();
+        int max_width = (res_values[size - 1].first != calibration_res.first) ? res_values[size - 1].first : res_values[size - 2].first;
+        int max_height = (res_values[size - 1].second != calibration_res.second) ? res_values[size - 1].second : res_values[size - 2].second;
```

Comment:
Need to be refactored - the `[..-1]/[..-2]` patterns may result in negative indexes

Reply:
following code added with relevant comment:
if (size <= 2)
            return false;